vignettes/projecting_incidence.Rmd

```diff
@@ -0,0 +1,292 @@
+serial_interval <- function(sample_size) {
+  si <- rlnorm(sample_size, meanlog = log_mean, sdlog = log_sd)
+  return(si)
+}
+```
+
+### Offspring distribution
+
+We will also assume that the offspring distribution is characterised by a 
+negative binomial with $\mathcal{R} = 2.5$ [@abbott2020] and 
+$\mathcal{k} = 0.58$ [@Wang2020]. In this parameterization, $\mathcal{R}$ 
+represents the $\mathcal{R_0}$, which is defined as the average number of 
+cases produced by a single individual in an entirely susceptible population. 
+The parameter $k$ represents superspreading, that is, the degree of 
+heterogeneity in transmission by single individuals.
+
+## Simulations
+To summarize the simulation set up, for each of the `r sim_rep` simulations,
+we want to project cases over a `r projection_window` day period since the 
+last case, assuming that no chain would exceed `r chain_threshold`. 
+
+### Model assumptions
+
+`chain_sim()` makes the following simplifying assumptions:
+
+1. All cases are observed
+1. There is no reporting delay
+1. Reporting rate is constant through the course of the epidemic
+1. No interventions have been implemented
+1. Population is homogeneous and well-mixed
+
+```{r simulations, message=FALSE}
+set.seed(1234)
+
+
+sim_chain_sizes <- lapply(
+  seq_len(sim_rep),
+  function(sim) {
+    chain_sim(
+      n = length(start_times),
+      offspring = "nbinom",
+      mu = 2.5,
+      size = 0.58,
+      stat = "size",
+      infinite = chain_threshold,
+      serial = serial_interval,
+      t0 = start_times,
+      tf = projection_end_day,
+      tree = TRUE
+    ) %>%
+      mutate(sim = sim)
+  }
+)
+
+sim_output <- bind_rows(sim_chain_sizes)
+
+head(sim_output)
+```
+
+## Post-processing
+
+From the simulated data, we will count the median daily cases 
+(`median_daily_cases`) across all simulations and overlay that on the results 
+of all the projections through time (`incidence_ts`).
+
+```{r post_processing}
+index_date <- min(covid19_sa$date)
+
+# Daily number of cases for each simulation
+incidence_ts <- sim_output %>%
+  mutate(day = ceiling(time)) %>%
+  group_by(sim, day) %>%
+  summarise(cases = n()) %>%
+  ungroup()
+
+# Add dates
+incidence_ts <- incidence_ts %>%
+  group_by(sim) %>%
+  mutate(date = index_date + days(seq(0, n() - 1))) %>%
+  ungroup()
+
+## Median daily number of cases aggregated across all simulations
+median_daily_cases <- incidence_ts %>%
+  group_by(day) %>%
+  summarise(median_cases = median(cases)) %>%
+  ungroup() %>%
+  arrange(day)
+
+# Add dates
+median_daily_cases <- median_daily_cases %>%
+  mutate(date = index_date + days(seq(0, projection_end_day))) %>%
+  ungroup()
+
+```
+
+
+## Visualization
+
+```{r viz, fig.cap ="COVID-19 incidence projected over a two week window. The gray lines represent individual simulations, red connected dots represent the median daily cases across all simulations, and the black triangles represent the observed data.", fig.width=2.0, fig.height=1.8}
```

Comment:
The figure looks squashed - perhaps experiment with increasing to e.g. `fig.width = 4.0, fig.height = 3.6`?

Also the individual trajectories are hard to see - perhaps only show a subset?

Reply:
I've increased the figure size and made the individual trajectories thinner.